frontend/src/components/Bundle/Bundle.js

```diff
@@ -62,114 +62,80 @@ class Bundle extends React.Component<
      * @return  jQuery Deferred object
      */
     fetchFileSummary(uuid, path) {
-        return $.ajax({
-            type: 'GET',
-            url: '/rest/bundles/' + uuid + '/contents/blob' + path,
-            data: {
-                head: 50,
-                tail: 50,
-                truncation_text: '\n... [truncated] ...\n\n',
-            },
-            dataType: 'text',
-            cache: false,
-            context: this, // automatically bind `this` in all callbacks
-        });
+        return fetchFileSummary(uuid, path);
     }
 
     /**
      * Fetch bundle data and update the state of this component.
      */
     refreshBundle = () => {
         // Fetch bundle metadata
-        $.ajax({
-            type: 'GET',
-
-            url: '/rest/bundles/' + this.props.uuid,
-            data: {
-                include_display_metadata: 1,
-                include: 'owner,group_permissions,host_worksheets',
-            },
-            dataType: 'json',
-            cache: false,
-            context: this, // automatically bind `this` in all callbacks
-        })
-            .then(function(response) {
-                // Normalize JSON API doc into simpler object
-                const bundleInfo = new JsonApiDataStore().sync(response);
-                bundleInfo.editableMetadataFields = response.data.meta.editable_metadata_keys;
-                bundleInfo.metadataType = response.data.meta.metadata_type;
-                this.setState({ bundleInfo: bundleInfo });
-            })
-            .fail(function(xhr, status, err) {
-                this.setState({
-                    bundleInfo: null,
-                    fileContents: null,
-                    stdout: null,
-                    stderr: null,
-                    errorMessages: this.state.errorMessages.concat([xhr.responseText]),
-                });
+        let callback = (response) => {
```

Comment:
Is there any value in it being a variable? Why not inline it inside the `.then()`

Reply:
I think it is cleaner this way otherwise the .then().catch() is going to be super long.

Reply:
Actually, the cleanest way to do this would be to use async-await. Then you could do something like this:

```
async refreshBundle = () => {
   try {
    const response = fetchBundleMetadata(this.props.uuid);
    const bundleInfo = new JsonApiDataStore().sync(response);
    // ... rest of callback here
   } catch (error) {
     if (error.response.status === 404) {
     // ... rest of error handler here
     }
   }
}
```

@jzwang43 , I was thinking it wasn't worth doing async-await for now so we could keep the changes as small as possible. We could probably do this in a later refactor, though.

Reply:
I think we should do it at a later time since this pull request is already very big.